### PR TITLE
WoodScape Tutorial version 0

### LINKED ISSUE #3: Create a complete getting started with an academic example
Maybe with woodscape if some models are trained on 
Or mnist if we find enough contextual data

### PR DESCRIPTION
**Pull request by SabrinaChaouche**

### CODE REVIEW

File: documentation/useCases/WoodScape_Tutorial/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Valeo WoodScape - 2D Objects Detection case study
+
+<br> 
+<br>
+
+The WoodScape dataset is a public dataset containing more than *100K
+images* of urban scenes captured using fish-eye cameras for automotive driving tasks. The images are provided by 4 different cameras with different angles of view (front, rear, middle right and middle left) with 360° coverage and have annotations for a diverse set of computer vision tasks.
+<br>
+In this case study, we used approximately *72K images* and focused on the 2D bounding boxes detection task with five classes: *vehicle, person, bicycle, traffic light and traffic sign.*
```
Comment (Tomansion) [inline documentation/useCases/WoodScape_Tutorial/README.md]: image
72k images or objects ?

File: documentation/useCases/WoodScape_Tutorial/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+The [Figure 1](#figure-01---the-parallel-coordinates-of-the-woodscape-dataset-before-filtering) displays the parallel coordinate of the WoodScape dataset filtered by the color of the *"camera_id"* values; while the [Figure 2](#figure-02---the-parallel-coordinates-of-the-woodscape-dataset-after-filtering-to-get-the-train-set-and-the-traffic-sign-class-data) shows some filters applied on the same parallel coordinate to select data. 
+
+##### Figure 01 - The Parallel coordinates of the WoodScape dataset before filtering. 
+![Parallel Coordinates 01](./images/01_Parallel_Coordinates.png)
+##### Figure 02 - The Parallel coordinates of the WoodScape dataset after filtering to get the train set and the traffic sign class data.
+![Parallel Coordinates 02](./images/02_Parallel_Coordinates.png)
+
+
+To further investigates the dataset, we explore the distribution of each one of the train, validation and test sets using the **“Data Distribution”** widget.
```
Comment (Tomansion) [inline documentation/useCases/WoodScape_Tutorial/README.md]: Nothing to learn from the two images ?

File: documentation/useCases/WoodScape_Tutorial/README.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+- Select some columns to explore their data by selecting values and trying to spot outliers and biases. 
+
+An interesting feature of DebiAI is the **“color button”**, which will discriminate the data according to the values of a given column, in our case, we choose the *“camera_id”* column which has 4 different values: *MVR, MVL, FV and RV*.
+
+The [Figure 1](#figure-01---the-parallel-coordinates-of-the-woodscape-dataset-before-filtering) displays the parallel coordinate of the WoodScape dataset filtered by the color of the *"camera_id"* values; while the [Figure 2](#figure-02---the-parallel-coordinates-of-the-woodscape-dataset-after-filtering-to-get-the-train-set-and-the-traffic-sign-class-data) shows some filters applied on the same parallel coordinate to select data. 
+
+##### Figure 01 - The Parallel coordinates of the WoodScape dataset before filtering. 
+![Parallel Coordinates 01](./images/01_Parallel_Coordinates.png)
+##### Figure 02 - The Parallel coordinates of the WoodScape dataset after filtering to get the train set and the traffic sign class data.
+![Parallel Coordinates 02](./images/02_Parallel_Coordinates.png)
```
Comment (Tomansion) [inline documentation/useCases/WoodScape_Tutorial/README.md]: I suggest that you move the images on top of the comments (just bellow the line n°26 or 28

File: documentation/useCases/WoodScape_Tutorial/README.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+
+To further investigates the dataset, we explore the distribution of each one of the train, validation and test sets using the **“Data Distribution”** widget.
+
+
+### Data Distribution 
+The [Figure 3](#figure-03---woodscape-dataset-split-from-left-to-train-train-validation-and-test-by-objects-class-distribution-grouped-by-camera-id) displays the distribution of each of the three sets by the class name grouped by the camera id. A first observation is the spectacular imbalance in the distribution of the five objects classes which is inherently related to the nature and the context of the task; in an urban road scene and in day time, it is normal to have more vehicles compared to traffic lights and traffic signs on the road. But it is important to perceive that we keep the same tendency / bias all over the three sets which is important for the accurate interpretation of the model’s outputs. The proportion of objects is more like homogenous across the four cameras in the three sets.
```
Comment (Tomansion) [inline documentation/useCases/WoodScape_Tutorial/README.md]: I suggest that you put the interpretation of the results bellow the image

Comment (Tomansion) [reply documentation/useCases/WoodScape_Tutorial/README.md]: Same suggestions for the next interpretations

File: documentation/useCases/WoodScape_Tutorial/README.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
+# Valeo WoodScape - 2D Objects Detection case study
+
+<br> 
+<br>
+
+The WoodScape dataset is a public dataset containing more than *100K
```
Comment (Tomansion) [inline documentation/useCases/WoodScape_Tutorial/README.md]: Please add a link to the Woodscape dataset

File: documentation/useCases/WoodScape_Tutorial/README.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+images* of urban scenes captured using fish-eye cameras for automotive driving tasks. The images are provided by 4 different cameras with different angles of view (front, rear, middle right and middle left) with 360° coverage and have annotations for a diverse set of computer vision tasks.
+<br>
+In this case study, we used approximately *72K images* and focused on the 2D bounding boxes detection task with five classes: *vehicle, person, bicycle, traffic light and traffic sign.*
+<br>
+<br>
+This tutorial will be divided into two distinct steps: <br>
+1. Exploring the dataset before the training process. <br>
+2. Analysing the models results.
```
Comment (Tomansion) [inline documentation/useCases/WoodScape_Tutorial/README.md]: 1. Please make those links to the different sections

2. Have you planned to do a `0. Inserting the project Data into DebiAI` step by step guide ?
    If yes, please create a second page with your documented python code or notebook (use a notebook to markdown tool). Then add in this page a link to your new page.

File: documentation/useCases/WoodScape_Tutorial/README.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+
+### Data Distribution 
+The [Figure 3](#figure-03---woodscape-dataset-split-from-left-to-train-train-validation-and-test-by-objects-class-distribution-grouped-by-camera-id) displays the distribution of each of the three sets by the class name grouped by the camera id. A first observation is the spectacular imbalance in the distribution of the five objects classes which is inherently related to the nature and the context of the task; in an urban road scene and in day time, it is normal to have more vehicles compared to traffic lights and traffic signs on the road. But it is important to perceive that we keep the same tendency / bias all over the three sets which is important for the accurate interpretation of the model’s outputs. The proportion of objects is more like homogenous across the four cameras in the three sets. 
+
+##### Figure 03 - WoodScape dataset split (from left to train: train, validation and test) by objects class distribution grouped by Camera ID
+![Data Distribution 03](./images/03_Data_Distribution.png)
```
Comment (Tomansion) [inline documentation/useCases/WoodScape_Tutorial/README.md]: I think that a step by step explanation of how to recreate this is required in this tutorial.

File: documentation/useCases/WoodScape_Tutorial/README.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+<br>
+
+## Results Analysis
+We used two versions of YOLO-based
```
Comment (Tomansion) [inline documentation/useCases/WoodScape_Tutorial/README.md]: Please add a link to learn more about Yolo

File: documentation/useCases/WoodScape_Tutorial/README.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+architectures, specifically YOLOv5 and YOLOv8. A YOLOv5 and a YOLOv8 models trained on WoodScape's train set and a YOLOv8 model trained on the COCO2017 dataset.
+
+We assess the performances of our models on the WoodScape's test set for each of the three models.
+
+
+### Night Stars Plot
+We use the **night stars plot widget** to display the relationship between the precision and the recall of each model to help us navigate the trade-off between the quality and the quantity of detections: are we favouring / prioritizing the safety or the availability of our detection system.
+
+The [Figure 4](#figure-05-the-relationship-between-the-precision-and-recall-for-each-model-using-the-night-stars-plot-widget) shows that the two models trained on the WoodScape's train set have better performances with one having better precision and the other showcasing better recall.
```
Comment (Tomansion) [inline documentation/useCases/WoodScape_Tutorial/README.md]: We don't see the X axis

File: documentation/useCases/WoodScape_Tutorial/README.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+Another question that arises is if the category of the objects influences the performance of the model. In order to explore this hypothesis, we choose two categories of objects *vehicles* and *persons* to study this possible correlation. <br>
+The [Figure 7](#figure-07---correlation-between-the-objects-categories-and-the-f1-score-left-f1-score-by-the-number-of-vehicles-right-f1-score-by-the-number-of-persons) displays the plots of the f1-score by the number of vehicles per image (left) and the number of persons per frame (right); a first observation is that the models' performances degrade as the number of persons increases, in the other hand, the performances improve as the number of vehicles grows in the images, which indicates that our models have a certain bias towards the vehicle category. This can be explained by the fact that our dataset has more objects belonging to this category compared to the four others.
+
+##### Figure 07 - Correlation between the objects' categories and the f1-score; left: f1-score by the number of vehicles, right: f1-score by the number of persons
+![Point Plot 07](./images/08_Point_Plot_f1-score_nb-vehicles_vs_f1-score_nb_persons.png)
```
Comment (Tomansion) [inline documentation/useCases/WoodScape_Tutorial/README.md]: I find the plot with the number of bike better in terms of model comparaison.
image

Moreover, bikes is a known challenge for Valeo

**Review by Tomansion — CHANGES_REQUESTED** [review]

File: documentation/useCases/WoodScape_Tutorial/README.md (hunk 6 above)
Comment (SabrinaChaouche) [reply documentation/useCases/WoodScape_Tutorial/README.md]: Reply to 2: this has been postponed.

File: documentation/useCases/WoodScape_Tutorial/README.md (hunk 9 above)
Comment (SabrinaChaouche) [reply documentation/useCases/WoodScape_Tutorial/README.md]: A problem to resolve on DebiAI!

File: documentation/useCases/WoodScape_Tutorial/README.md [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+The [WoodScape](https://woodscape.valeo.com/woodscape/) dataset is a public dataset containing more than *100K images* of urban scenes captured using fish-eye cameras for automotive driving tasks. The images are provided by 4 different cameras with different angles of view (front, rear, middle right and middle left) with 360° coverage and have annotations for a diverse set of computer vision tasks.
+<br>
+In this case study, we used *8.234 images* (divided into three sets: train, validation and test) and we focused on the 2D bounding boxes detection task with five classes: *vehicle, person, bicycle, traffic light and traffic sign.*
+
+<br>
+
+This tutorial will be divided into two distinct sections, each one ties to a specific project in the DebiAI Demo Instance:
+
+1. Exploring the dataset before the training process in the project *WoodScape - Objects*. [See here](#dataset-exploration)
+2. Analysing the models results in the project *WoodScape - Images*. [See here](#results-analysis)
```
Comment (Tomansion) [inline documentation/useCases/WoodScape_Tutorial/README.md]: The two links don't work

**Review by Tomansion — APPROVED** [review]

**Pull request merged**